src/test/java/org/opensearch/knn/jni/JNIServiceTests.java

```diff
@@ -590,12 +590,12 @@ public void testLoadIndex_faiss_valid() throws IOException {
     }
 
     public void testQueryIndex_invalidEngine() {
-        expectThrows(IllegalArgumentException.class, () -> JNIService.queryIndex(0L, new float[] {}, 0, "invalid" + "-engine", null));
```

Comment:
Should we add a test for parent filter with faiss?

Reply:
We have it.

Reply:
Thats a restful integ test though. Should we unit test the functionality as well?

Reply:
Unit test is added in `KNNWeightTests.java`

Reply:
You mean for this JNIServiceTests case? I think it will be covered by unit test in JNI side but let me add same one in java side as well.